src/main/java/org/openrewrite/java/testing/assertj/JUnitAssertFalseToAssertThat.java

```diff
@@ -44,66 +46,37 @@ public String getDescription() {
-            if (assertionsParser == null) {
-                assertionsParser = JavaParser.fromJavaVersion()
-                        .classpathFromResources(ctx, "assertj-core-3.24");
-            }
-            return assertionsParser;
-        }
-
-        private static final MethodMatcher JUNIT_ASSERT_FALSE = new MethodMatcher("org.junit.jupiter.api.Assertions" + " assertFalse(boolean, ..)");
-
-        @Override
-        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
-            if (!JUNIT_ASSERT_FALSE.matches(method)) {
-                return method;
-            }
+        return Preconditions.check(new UsesMethod<>(ASSERT_FALSE_MATCHER), new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                J.MethodInvocation mi = super.visitMethodInvocation(method, ctx);
+                if (!ASSERT_FALSE_MATCHER.matches(mi)) {
+                    return mi;
+                }
+
+                maybeAddImport("org.assertj.core.api.Assertions", "assertThat");
+                maybeRemoveImport("org.junit.jupiter.api.Assertions");
+
+                List<Expression> args = mi.getArguments();
+                Expression actual = args.get(0);
+                if (args.size() == 1) {
+                    return JavaTemplate.builder("assertThat(#{any(boolean)}).isFalse();")
+                            .staticImports("org.assertj.core.api.Assertions.assertThat")
+                            .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "assertj-core-3.24"))
+                            .build()
+                            .apply(getCursor(), mi.getCoordinates().replace(), actual);
+                }
 
-            List<Expression> args = method.getArguments();
-            Expression actual = args.get(0);
-
-            if (args.size() == 1) {
-                method = JavaTemplate.builder("assertThat(#{any(boolean)}).isFalse();")
-                        .staticImports("org.assertj.core.api.Assertions.assertThat")
-                        .javaParser(assertionsParser(ctx))
-                        .build()
-                        .apply(
-                                getCursor(),
-                                method.getCoordinates().replace(),
-                                actual
-                        );
-            } else {
                 Expression message = args.get(1);
                 JavaTemplate.Builder template = TypeUtils.isString(message.getType()) ?
                         JavaTemplate.builder("assertThat(#{any(boolean)}).as(#{any(String)}).isFalse();") :
                         JavaTemplate.builder("assertThat(#{any(boolean)}).as(#{any(java.util.function.Supplier)}).isFalse();");
-
-                method = template
+                return template
```

Comment:
As seen on JUnitAssertInstanceOfToAssertThat there's no need for a ternary here or explicit type in that `any()` when reusing existing LST elements. The type will be deduced and can reduce the logic needed here.

https://github.com/openrewrite/rewrite-testing-frameworks/blob/152617c99daaca8384f3c4376ce183fdfd3a4268/src/main/java/org/openrewrite/java/testing/assertj/JUnitAssertInstanceOfToAssertThat.java#L67-L68